<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title id="head-title">report.html</title>
      <style type="text/css">body {
  font-family: Helvetica, Arial, sans-serif;
  font-size: 12px;
  /* do not increase min-width as some may use split screens */
  min-width: 800px;
  color: #999;
}

h1 {
  font-size: 24px;
  color: black;
}

h2 {
  font-size: 16px;
  color: black;
}

p {
  color: black;
}

a {
  color: #999;
}

table {
  border-collapse: collapse;
}

/******************************
 * SUMMARY INFORMATION
 ******************************/
#environment td {
  padding: 5px;
  border: 1px solid #e6e6e6;
  vertical-align: top;
}
#environment tr:nth-child(odd) {
  background-color: #f6f6f6;
}
#environment ul {
  margin: 0;
  padding: 0 20px;
}

/******************************
 * TEST RESULT COLORS
 ******************************/
span.passed,
.passed .col-result {
  color: green;
}

span.skipped,
span.xfailed,
span.rerun,
.skipped .col-result,
.xfailed .col-result,
.rerun .col-result {
  color: orange;
}

span.error,
span.failed,
span.xpassed,
.error .col-result,
.failed .col-result,
.xpassed .col-result {
  color: red;
}

.col-links__extra {
  margin-right: 3px;
}

/******************************
 * RESULTS TABLE
 *
 * 1. Table Layout
 * 2. Extra
 * 3. Sorting items
 *
 ******************************/
/*------------------
 * 1. Table Layout
 *------------------*/
#results-table {
  border: 1px solid #e6e6e6;
  color: #999;
  font-size: 12px;
  width: 100%;
}
#results-table th,
#results-table td {
  padding: 5px;
  border: 1px solid #e6e6e6;
  text-align: left;
}
#results-table th {
  font-weight: bold;
}

/*------------------
 * 2. Extra
 *------------------*/
.logwrapper {
  max-height: 230px;
  overflow-y: scroll;
  background-color: #e6e6e6;
}
.logwrapper.expanded {
  max-height: none;
}
.logwrapper.expanded .logexpander:after {
  content: "collapse [-]";
}
.logwrapper .logexpander {
  z-index: 1;
  position: sticky;
  top: 10px;
  width: max-content;
  border: 1px solid;
  border-radius: 3px;
  padding: 5px 7px;
  margin: 10px 0 10px calc(100% - 80px);
  cursor: pointer;
  background-color: #e6e6e6;
}
.logwrapper .logexpander:after {
  content: "expand [+]";
}
.logwrapper .logexpander:hover {
  color: #000;
  border-color: #000;
}
.logwrapper .log {
  min-height: 40px;
  position: relative;
  top: -50px;
  height: calc(100% + 50px);
  border: 1px solid #e6e6e6;
  color: black;
  display: block;
  font-family: "Courier New", Courier, monospace;
  padding: 5px;
  padding-right: 80px;
  white-space: pre-wrap;
}

div.media {
  border: 1px solid #e6e6e6;
  float: right;
  height: 240px;
  margin: 0 5px;
  overflow: hidden;
  width: 320px;
}

.media-container {
  display: grid;
  grid-template-columns: 25px auto 25px;
  align-items: center;
  flex: 1 1;
  overflow: hidden;
  height: 200px;
}

.media-container--fullscreen {
  grid-template-columns: 0px auto 0px;
}

.media-container__nav--right,
.media-container__nav--left {
  text-align: center;
  cursor: pointer;
}

.media-container__viewport {
  cursor: pointer;
  text-align: center;
  height: inherit;
}
.media-container__viewport img,
.media-container__viewport video {
  object-fit: cover;
  width: 100%;
  max-height: 100%;
}

.media__name,
.media__counter {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  flex: 0 0 25px;
  align-items: center;
}

.collapsible td:not(.col-links) {
  cursor: pointer;
}
.collapsible td:not(.col-links):hover::after {
  color: #bbb;
  font-style: italic;
  cursor: pointer;
}

.col-result {
  width: 130px;
}
.col-result:hover::after {
  content: " (hide details)";
}

.col-result.collapsed:hover::after {
  content: " (show details)";
}

#environment-header h2:hover::after {
  content: " (hide details)";
  color: #bbb;
  font-style: italic;
  cursor: pointer;
  font-size: 12px;
}

#environment-header.collapsed h2:hover::after {
  content: " (show details)";
  color: #bbb;
  font-style: italic;
  cursor: pointer;
  font-size: 12px;
}

/*------------------
 * 3. Sorting items
 *------------------*/
.sortable {
  cursor: pointer;
}
.sortable.desc:after {
  content: " ";
  position: relative;
  left: 5px;
  bottom: -12.5px;
  border: 10px solid #4caf50;
  border-bottom: 0;
  border-left-color: transparent;
  border-right-color: transparent;
}
.sortable.asc:after {
  content: " ";
  position: relative;
  left: 5px;
  bottom: 12.5px;
  border: 10px solid #4caf50;
  border-top: 0;
  border-left-color: transparent;
  border-right-color: transparent;
}

.hidden, .summary__reload__button.hidden {
  display: none;
}

.summary__data {
  flex: 0 0 550px;
}
.summary__reload {
  flex: 1 1;
  display: flex;
  justify-content: center;
}
.summary__reload__button {
  flex: 0 0 300px;
  display: flex;
  color: white;
  font-weight: bold;
  background-color: #4caf50;
  text-align: center;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  cursor: pointer;
}
.summary__reload__button:hover {
  background-color: #46a049;
}
.summary__spacer {
  flex: 0 0 550px;
}

.controls {
  display: flex;
  justify-content: space-between;
}

.filters,
.collapse {
  display: flex;
  align-items: center;
}
.filters button,
.collapse button {
  color: #999;
  border: none;
  background: none;
  cursor: pointer;
  text-decoration: underline;
}
.filters button:hover,
.collapse button:hover {
  color: #ccc;
}

.filter__label {
  margin-right: 10px;
}

      </style>
    
  </head>
  <body>
    <h1 id="title">report.html</h1>
    <p>Report generated on 06-Aug-2025 at 09:07:13 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a>
        v4.1.1</p>
    <div id="environment-header">
      <h2>Environment</h2>
    </div>
    <table id="environment"></table>
    <!-- TEMPLATES -->
      <template id="template_environment_row">
      <tr>
        <td></td>
        <td></td>
      </tr>
    </template>
    <template id="template_results-table__body--empty">
      <tbody class="results-table-row">
        <tr id="not-found-message">
          <td colspan="4">No results found. Check the filters.</th>
        </tr>
    </template>
    <template id="template_results-table__tbody">
      <tbody class="results-table-row">
        <tr class="collapsible">
        </tr>
        <tr class="extras-row">
          <td class="extra" colspan="4">
            <div class="extraHTML"></div>
            <div class="media">
              <div class="media-container">
                  <div class="media-container__nav--left"><</div>
                  <div class="media-container__viewport">
                    <img src="" />
                    <video controls>
                      <source src="" type="video/mp4">
                    </video>
                  </div>
                  <div class="media-container__nav--right">></div>
                </div>
                <div class="media__name"></div>
                <div class="media__counter"></div>
            </div>
            <div class="logwrapper">
              <div class="logexpander"></div>
              <div class="log"></div>
            </div>
          </td>
        </tr>
      </tbody>
    </template>
    <!-- END TEMPLATES -->
    <div class="summary">
      <div class="summary__data">
        <h2>Summary</h2>
        <div class="additional-summary prefix">
        </div>
        <p class="run-count">0 test took 00:01:45.</p>
        <p class="filter">(Un)check the boxes to filter the results.</p>
        <div class="summary__reload">
          <div class="summary__reload__button hidden" onclick="location.reload()">
            <div>There are still tests running. <br />Reload this page to get the latest results!</div>
          </div>
        </div>
        <div class="summary__spacer"></div>
        <div class="controls">
          <div class="filters">
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="failed" disabled/>
            <span class="failed">0 Failed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="passed" disabled/>
            <span class="passed">0 Passed,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="skipped" />
            <span class="skipped">1 Skipped,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xfailed" disabled/>
            <span class="xfailed">0 Expected failures,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="xpassed" disabled/>
            <span class="xpassed">0 Unexpected passes,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="error" disabled/>
            <span class="error">0 Errors,</span>
            <input checked="true" class="filter" name="filter_checkbox" type="checkbox" data-test-result="rerun" disabled/>
            <span class="rerun">0 Reruns</span>
          </div>
          <div class="collapse">
            <button id="show_all_details">Show all details</button>&nbsp;/&nbsp;<button id="hide_all_details">Hide all details</button>
          </div>
        </div>
      </div>
      <div class="additional-summary summary">
      </div>
      <div class="additional-summary postfix">
      </div>
    </div>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable" data-column-type="result">Result</th>
          <th class="sortable" data-column-type="testId">Test</th>
          <th class="sortable" data-column-type="duration">Duration</th>
          <th>Links</th>
        </tr>
      </thead>
    </table>
  </body>
  <footer>
    <div id="data-container" data-jsonblob="{&#34;environment&#34;: {&#34;Python&#34;: &#34;3.13.0&#34;, &#34;Platform&#34;: &#34;Windows-11-10.0.26100-SP0&#34;, &#34;Packages&#34;: {&#34;pytest&#34;: &#34;7.4.3&#34;, &#34;pluggy&#34;: &#34;1.6.0&#34;}, &#34;Plugins&#34;: {&#34;html&#34;: &#34;4.1.1&#34;, &#34;json-report&#34;: &#34;1.5.0&#34;, &#34;metadata&#34;: &#34;3.1.1&#34;}, &#34;JAVA_HOME&#34;: &#34;C:\\Users\\E017093\\AppData\\Local\\Programs\\Microsoft\\jdk-21.0.8.9-hotspot\\&#34;}, &#34;tests&#34;: {&#34;tests/moduletest/test_discovery.py::TestDiscovery::test_discover_project_spaces&#34;: [{&#34;extras&#34;: [], &#34;result&#34;: &#34;Skipped&#34;, &#34;testId&#34;: &#34;tests/moduletest/test_discovery.py::TestDiscovery::test_discover_project_spaces&#34;, &#34;duration&#34;: &#34;00:01:45&#34;, &#34;resultsTableRow&#34;: [&#34;&lt;td class=\&#34;col-result\&#34;&gt;Skipped&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-testId\&#34;&gt;tests/moduletest/test_discovery.py::TestDiscovery::test_discover_project_spaces&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-duration\&#34;&gt;00:01:45&lt;/td&gt;&#34;, &#34;&lt;td class=\&#34;col-links\&#34;&gt;&lt;/td&gt;&#34;], &#34;log&#34;: &#34;(&amp;#x27;C:\\\\Users\\\\E017093\\\\gitRepos\\\\PolarionMock\\\\tests\\\\moduletest\\\\test_discovery.py&amp;#x27;, 48, &amp;#x27;Skipped: No spaces found in project (might be empty project)&amp;#x27;)\n\n------------------------------ Captured log setup ------------------------------\nINFO     tests.session:conftest.py:231 ================================================================================\nINFO     tests.session:conftest.py:232 Test Session Started: 20250806_090529\nINFO     tests.session:conftest.py:233 Environment: production\nINFO     tests.session:conftest.py:234 ================================================================================\nDEBUG    tests.session:conftest.py:151 Test environment: production\nDEBUG    tests.session:conftest.py:152 Base URL: N/A\nINFO     polarion_api.client:client.py:60 Initialized Polarion client for https://polarion-d.claas.local\n\n------------------------------ Captured log call -------------------------------\nINFO     polarion_api.documents:documents.py:398 Starting document and space discovery for project: Python\nDEBUG    polarion_api.documents:documents.py:415 Fetching documents page 1 for project Python\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (1): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555BA10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (2): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555EFD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (3): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555F250&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (4): polarion-d.claas.local:443\nERROR    polarion_api.documents:documents.py:485 Error during document discovery: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/documents?page%5Bsize%5D=100&amp;amp;page%5Bnumber%5D=1&amp;amp;fields%5Bdocuments%5D=id%2Ctitle%2Ctype%2CmoduleURI%2CspaceId (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555F4D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nINFO     polarion_api.documents:documents.py:513 Using fallback space discovery method\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/_default/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (5): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555FB10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (6): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555FD90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (7): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296878050&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (8): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;_default&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/_default/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/_default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968782D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Default/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (9): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296878910&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (10): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296878B90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (11): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879090&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (12): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Default&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Default/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879310&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/default/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (13): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296878E10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (14): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968796D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (15): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879950&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (16): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;default&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/default/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/default/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879BD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Requirements/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (17): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879F90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (18): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687A210&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (19): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687A490&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (20): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Requirements&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Requirements/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Requirements/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687A710&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/TestCases/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (21): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687AAD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (22): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687AD50&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (23): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687AFD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (24): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;TestCases&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/TestCases/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/TestCases/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687B250&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Documents/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (25): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687B610&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (26): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687B890&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (27): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687BB10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (28): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Documents&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Documents/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Documents/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687BD90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Specifications/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (29): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E8190&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (30): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687BC50&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (31): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687B110&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (32): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Specifications&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Specifications/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Specifications/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029687A5D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Tests/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (33): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020296879310&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (34): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968782D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (35): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x000002029555D810&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (36): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Tests&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Tests/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Tests/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E82D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Design/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (37): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E8690&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (38): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E8910&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (39): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E8B90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (40): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Design&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Design/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Design/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E8E10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Implementation/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (41): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E91D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (42): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E9450&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (43): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E96D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (44): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Implementation&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Implementation/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Implementation/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E9950&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Maintenance/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (45): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E9D10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (46): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968E9F90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (47): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EA210&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (48): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Maintenance&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Maintenance/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Maintenance/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EA490&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Product Layer/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (49): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EA850&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (50): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EAAD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (51): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EAD50&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (52): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Product Layer&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Product Layer/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Product%20Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EAFD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/ProductLayer/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (53): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EB390&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (54): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EB610&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (55): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EB890&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (56): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;ProductLayer&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/ProductLayer/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/ProductLayer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EBB10&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/product_layer/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (57): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EBED0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (58): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670190&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (59): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670410&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (60): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;product_layer&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/product_layer/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/product_layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670690&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Product-Layer/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (61): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670A50&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (62): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670CD0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (63): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295670F50&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (64): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;Product-Layer&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/Product-Layer/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/Product-Layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202956711D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nDEBUG    polarion_api.client:client.py:107 GET https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/product-layer/documents\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (65): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=2, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295671590&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (66): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=1, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x0000020295671810&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (67): polarion-d.claas.local:443\nDEBUG    urllib3.util.retry:retry.py:521 Incremented Retry for (url=&amp;#x27;/polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle&amp;#x27;): Retry(total=0, connect=None, read=None, redirect=None, status=None)\nWARNING  urllib3.connectionpool:connectionpool.py:868 Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &amp;#x27;NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EB9D0&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;)&amp;#x27;: /polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle\nDEBUG    urllib3.connectionpool:connectionpool.py:1049 Starting new HTTPS connection (68): polarion-d.claas.local:443\nDEBUG    polarion_api.documents:documents.py:555 Space &amp;#x27;product-layer&amp;#x27; not accessible: Failed to connect to https://polarion-d.claas.local/polarion/rest/projects/Python/spaces/product-layer/documents: HTTPSConnectionPool(host=&amp;#x27;polarion-d.claas.local&amp;#x27;, port=443): Max retries exceeded with url: /polarion/rest/projects/Python/spaces/product-layer/documents?page%5Bsize%5D=10&amp;amp;fields%5Bdocuments%5D=id%2Ctitle (Caused by NameResolutionError(&amp;quot;&amp;lt;urllib3.connection.HTTPSConnection object at 0x00000202968EAE90&amp;gt;: Failed to resolve &amp;#x27;polarion-d.claas.local&amp;#x27; ([Errno 11001] getaddrinfo failed)&amp;quot;))\nINFO     polarion_api.documents:documents.py:502 Discovery completed: Found 0 spaces and 0 documents\n\n&#34;}]}, &#34;renderCollapsed&#34;: [&#34;passed&#34;], &#34;initialSort&#34;: &#34;result&#34;, &#34;title&#34;: &#34;report.html&#34;}"></div>
    <script>
      (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
const { getCollapsedCategory, setCollapsedIds } = require('./storage.js')

class DataManager {
    setManager(data) {
        const collapsedCategories = [...getCollapsedCategory(data.renderCollapsed)]
        const collapsedIds = []
        const tests = Object.values(data.tests).flat().map((test, index) => {
            const collapsed = collapsedCategories.includes(test.result.toLowerCase())
            const id = `test_${index}`
            if (collapsed) {
                collapsedIds.push(id)
            }
            return {
                ...test,
                id,
                collapsed,
            }
        })
        const dataBlob = { ...data, tests }
        this.data = { ...dataBlob }
        this.renderData = { ...dataBlob }
        setCollapsedIds(collapsedIds)
    }

    get allData() {
        return { ...this.data }
    }

    resetRender() {
        this.renderData = { ...this.data }
    }

    setRender(data) {
        this.renderData.tests = [...data]
    }

    toggleCollapsedItem(id) {
        this.renderData.tests = this.renderData.tests.map((test) =>
            test.id === id ? { ...test, collapsed: !test.collapsed } : test,
        )
    }

    set allCollapsed(collapsed) {
        this.renderData = { ...this.renderData, tests: [...this.renderData.tests.map((test) => (
            { ...test, collapsed }
        ))] }
    }

    get testSubset() {
        return [...this.renderData.tests]
    }

    get environment() {
        return this.renderData.environment
    }

    get initialSort() {
        return this.data.initialSort
    }
}

module.exports = {
    manager: new DataManager(),
}

},{"./storage.js":8}],2:[function(require,module,exports){
const mediaViewer = require('./mediaviewer.js')
const templateEnvRow = document.getElementById('template_environment_row')
const templateResult = document.getElementById('template_results-table__tbody')

function htmlToElements(html) {
    const temp = document.createElement('template')
    temp.innerHTML = html
    return temp.content.childNodes
}

const find = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return elem.querySelector(selector)
}

const findAll = (selector, elem) => {
    if (!elem) {
        elem = document
    }
    return [...elem.querySelectorAll(selector)]
}

const dom = {
    getStaticRow: (key, value) => {
        const envRow = templateEnvRow.content.cloneNode(true)
        const isObj = typeof value === 'object' && value !== null
        const values = isObj ? Object.keys(value).map((k) => `${k}: ${value[k]}`) : null

        const valuesElement = htmlToElements(
            values ? `<ul>${values.map((val) => `<li>${val}</li>`).join('')}<ul>` : `<div>${value}</div>`)[0]
        const td = findAll('td', envRow)
        td[0].textContent = key
        td[1].appendChild(valuesElement)

        return envRow
    },
    getResultTBody: ({ testId, id, log, extras, resultsTableRow, tableHtml, result, collapsed }) => {
        const resultBody = templateResult.content.cloneNode(true)
        resultBody.querySelector('tbody').classList.add(result.toLowerCase())
        resultBody.querySelector('tbody').id = testId
        resultBody.querySelector('.collapsible').dataset.id = id

        resultsTableRow.forEach((html) => {
            const t = document.createElement('template')
            t.innerHTML = html
            resultBody.querySelector('.collapsible').appendChild(t.content)
        })

        if (log) {
            // Wrap lines starting with "E" with span.error to color those lines red
            const wrappedLog = log.replace(/^E.*$/gm, (match) => `<span class="error">${match}</span>`)
            resultBody.querySelector('.log').innerHTML = wrappedLog
        } else {
            resultBody.querySelector('.log').remove()
        }

        if (collapsed) {
            resultBody.querySelector('.collapsible > td')?.classList.add('collapsed')
            resultBody.querySelector('.extras-row').classList.add('hidden')
        } else {
            resultBody.querySelector('.collapsible > td')?.classList.remove('collapsed')
        }

        const media = []
        extras?.forEach(({ name, format_type, content }) => {
            if (['image', 'video'].includes(format_type)) {
                media.push({ path: content, name, format_type })
            }

            if (format_type === 'html') {
                resultBody.querySelector('.extraHTML').insertAdjacentHTML('beforeend', `<div>${content}</div>`)
            }
        })
        mediaViewer.setup(resultBody, media)

        // Add custom html from the pytest_html_results_table_html hook
        tableHtml?.forEach((item) => {
            resultBody.querySelector('td[class="extra"]').insertAdjacentHTML('beforeend', item)
        })

        return resultBody
    },
}

module.exports = {
    dom,
    htmlToElements,
    find,
    findAll,
}

},{"./mediaviewer.js":6}],3:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const storageModule = require('./storage.js')

const getFilteredSubSet = (filter) =>
    manager.allData.tests.filter(({ result }) => filter.includes(result.toLowerCase()))

const doInitFilter = () => {
    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)
}

const doFilter = (type, show) => {
    if (show) {
        storageModule.showCategory(type)
    } else {
        storageModule.hideCategory(type)
    }

    const currentFilter = storageModule.getVisible()
    const filteredSubset = getFilteredSubSet(currentFilter)
    manager.setRender(filteredSubset)

    const sortColumn = storageModule.getSort()
    doSort(sortColumn, true)
}

module.exports = {
    doFilter,
    doInitFilter,
}

},{"./datamanager.js":1,"./sort.js":7,"./storage.js":8}],4:[function(require,module,exports){
const { redraw, bindEvents, renderStatic } = require('./main.js')
const { doInitFilter } = require('./filter.js')
const { doInitSort } = require('./sort.js')
const { manager } = require('./datamanager.js')
const data = JSON.parse(document.getElementById('data-container').dataset.jsonblob)

function init() {
    manager.setManager(data)
    doInitFilter()
    doInitSort()
    renderStatic()
    redraw()
    bindEvents()
}

init()

},{"./datamanager.js":1,"./filter.js":3,"./main.js":5,"./sort.js":7}],5:[function(require,module,exports){
const { dom, find, findAll } = require('./dom.js')
const { manager } = require('./datamanager.js')
const { doSort } = require('./sort.js')
const { doFilter } = require('./filter.js')
const {
    getVisible,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    getSortDirection,
    possibleFilters,
} = require('./storage.js')

const removeChildren = (node) => {
    while (node.firstChild) {
        node.removeChild(node.firstChild)
    }
}

const renderStatic = () => {
    const renderEnvironmentTable = () => {
        const environment = manager.environment
        const rows = Object.keys(environment).map((key) => dom.getStaticRow(key, environment[key]))
        const table = document.getElementById('environment')
        removeChildren(table)
        rows.forEach((row) => table.appendChild(row))
    }
    renderEnvironmentTable()
}

const addItemToggleListener = (elem) => {
    elem.addEventListener('click', ({ target }) => {
        const id = target.parentElement.dataset.id
        manager.toggleCollapsedItem(id)

        const collapsedIds = getCollapsedIds()
        if (collapsedIds.includes(id)) {
            const updated = collapsedIds.filter((item) => item !== id)
            setCollapsedIds(updated)
        } else {
            collapsedIds.push(id)
            setCollapsedIds(collapsedIds)
        }
        redraw()
    })
}

const renderContent = (tests) => {
    const sortAttr = getSort(manager.initialSort)
    const sortAsc = JSON.parse(getSortDirection())
    const rows = tests.map(dom.getResultTBody)
    const table = document.getElementById('results-table')
    const tableHeader = document.getElementById('results-table-head')

    const newTable = document.createElement('table')
    newTable.id = 'results-table'

    // remove all sorting classes and set the relevant
    findAll('.sortable', tableHeader).forEach((elem) => elem.classList.remove('asc', 'desc'))
    tableHeader.querySelector(`.sortable[data-column-type="${sortAttr}"]`)?.classList.add(sortAsc ? 'desc' : 'asc')
    newTable.appendChild(tableHeader)

    if (!rows.length) {
        const emptyTable = document.getElementById('template_results-table__body--empty').content.cloneNode(true)
        newTable.appendChild(emptyTable)
    } else {
        rows.forEach((row) => {
            if (!!row) {
                findAll('.collapsible td:not(.col-links', row).forEach(addItemToggleListener)
                find('.logexpander', row).addEventListener('click',
                    (evt) => evt.target.parentNode.classList.toggle('expanded'),
                )
                newTable.appendChild(row)
            }
        })
    }

    table.replaceWith(newTable)
}

const renderDerived = () => {
    const currentFilter = getVisible()
    possibleFilters.forEach((result) => {
        const input = document.querySelector(`input[data-test-result="${result}"]`)
        input.checked = currentFilter.includes(result)
    })
}

const bindEvents = () => {
    const filterColumn = (evt) => {
        const { target: element } = evt
        const { testResult } = element.dataset

        doFilter(testResult, element.checked)
        const collapsedIds = getCollapsedIds()
        const updated = manager.renderData.tests.map((test) => {
            return {
                ...test,
                collapsed: collapsedIds.includes(test.id),
            }
        })
        manager.setRender(updated)
        redraw()
    }

    const header = document.getElementById('environment-header')
    header.addEventListener('click', () => {
        const table = document.getElementById('environment')
        table.classList.toggle('hidden')
        header.classList.toggle('collapsed')
    })

    findAll('input[name="filter_checkbox"]').forEach((elem) => {
        elem.addEventListener('click', filterColumn)
    })

    findAll('.sortable').forEach((elem) => {
        elem.addEventListener('click', (evt) => {
            const { target: element } = evt
            const { columnType } = element.dataset
            doSort(columnType)
            redraw()
        })
    })

    document.getElementById('show_all_details').addEventListener('click', () => {
        manager.allCollapsed = false
        setCollapsedIds([])
        redraw()
    })
    document.getElementById('hide_all_details').addEventListener('click', () => {
        manager.allCollapsed = true
        const allIds = manager.renderData.tests.map((test) => test.id)
        setCollapsedIds(allIds)
        redraw()
    })
}

const redraw = () => {
    const { testSubset } = manager

    renderContent(testSubset)
    renderDerived()
}

module.exports = {
    redraw,
    bindEvents,
    renderStatic,
}

},{"./datamanager.js":1,"./dom.js":2,"./filter.js":3,"./sort.js":7,"./storage.js":8}],6:[function(require,module,exports){
class MediaViewer {
    constructor(assets) {
        this.assets = assets
        this.index = 0
    }

    nextActive() {
        this.index = this.index === this.assets.length - 1 ? 0 : this.index + 1
        return [this.activeFile, this.index]
    }

    prevActive() {
        this.index = this.index === 0 ? this.assets.length - 1 : this.index -1
        return [this.activeFile, this.index]
    }

    get currentIndex() {
        return this.index
    }

    get activeFile() {
        return this.assets[this.index]
    }
}


const setup = (resultBody, assets) => {
    if (!assets.length) {
        resultBody.querySelector('.media').classList.add('hidden')
        return
    }

    const mediaViewer = new MediaViewer(assets)
    const container = resultBody.querySelector('.media-container')
    const leftArrow = resultBody.querySelector('.media-container__nav--left')
    const rightArrow = resultBody.querySelector('.media-container__nav--right')
    const mediaName = resultBody.querySelector('.media__name')
    const counter = resultBody.querySelector('.media__counter')
    const imageEl = resultBody.querySelector('img')
    const sourceEl = resultBody.querySelector('source')
    const videoEl = resultBody.querySelector('video')

    const setImg = (media, index) => {
        if (media?.format_type === 'image') {
            imageEl.src = media.path

            imageEl.classList.remove('hidden')
            videoEl.classList.add('hidden')
        } else if (media?.format_type === 'video') {
            sourceEl.src = media.path

            videoEl.classList.remove('hidden')
            imageEl.classList.add('hidden')
        }

        mediaName.innerText = media?.name
        counter.innerText = `${index + 1} / ${assets.length}`
    }
    setImg(mediaViewer.activeFile, mediaViewer.currentIndex)

    const moveLeft = () => {
        const [media, index] = mediaViewer.prevActive()
        setImg(media, index)
    }
    const doRight = () => {
        const [media, index] = mediaViewer.nextActive()
        setImg(media, index)
    }
    const openImg = () => {
        window.open(mediaViewer.activeFile.path, '_blank')
    }
    if (assets.length === 1) {
        container.classList.add('media-container--fullscreen')
    } else {
        leftArrow.addEventListener('click', moveLeft)
        rightArrow.addEventListener('click', doRight)
    }
    imageEl.addEventListener('click', openImg)
}

module.exports = {
    setup,
}

},{}],7:[function(require,module,exports){
const { manager } = require('./datamanager.js')
const storageModule = require('./storage.js')

const genericSort = (list, key, ascending, customOrder) => {
    let sorted
    if (customOrder) {
        sorted = list.sort((a, b) => {
            const aValue = a.result.toLowerCase()
            const bValue = b.result.toLowerCase()

            const aIndex = customOrder.findIndex((item) => item.toLowerCase() === aValue)
            const bIndex = customOrder.findIndex((item) => item.toLowerCase() === bValue)

            // Compare the indices to determine the sort order
            return aIndex - bIndex
        })
    } else {
        sorted = list.sort((a, b) => a[key] === b[key] ? 0 : a[key] > b[key] ? 1 : -1)
    }

    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const durationSort = (list, ascending) => {
    const parseDuration = (duration) => {
        if (duration.includes(':')) {
            // If it's in the format "HH:mm:ss"
            const [hours, minutes, seconds] = duration.split(':').map(Number)
            return (hours * 3600 + minutes * 60 + seconds) * 1000
        } else {
            // If it's in the format "nnn ms"
            return parseInt(duration)
        }
    }
    const sorted = list.sort((a, b) => parseDuration(a['duration']) - parseDuration(b['duration']))
    if (ascending) {
        sorted.reverse()
    }
    return sorted
}

const doInitSort = () => {
    const type = storageModule.getSort(manager.initialSort)
    const ascending = storageModule.getSortDirection()
    const list = manager.testSubset
    const initialOrder = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed', 'Skipped', 'Passed']

    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    if (type?.toLowerCase() === 'original') {
        manager.setRender(list)
    } else {
        let sortedList
        switch (type) {
        case 'duration':
            sortedList = durationSort(list, ascending)
            break
        case 'result':
            sortedList = genericSort(list, type, ascending, initialOrder)
            break
        default:
            sortedList = genericSort(list, type, ascending)
            break
        }
        manager.setRender(sortedList)
    }
}

const doSort = (type, skipDirection) => {
    const newSortType = storageModule.getSort(manager.initialSort) !== type
    const currentAsc = storageModule.getSortDirection()
    let ascending
    if (skipDirection) {
        ascending = currentAsc
    } else {
        ascending = newSortType ? false : !currentAsc
    }
    storageModule.setSort(type)
    storageModule.setSortDirection(ascending)

    const list = manager.testSubset
    const sortedList = type === 'duration' ? durationSort(list, ascending) : genericSort(list, type, ascending)
    manager.setRender(sortedList)
}

module.exports = {
    doInitSort,
    doSort,
}

},{"./datamanager.js":1,"./storage.js":8}],8:[function(require,module,exports){
const possibleFilters = [
    'passed',
    'skipped',
    'failed',
    'error',
    'xfailed',
    'xpassed',
    'rerun',
]

const getVisible = () => {
    const url = new URL(window.location.href)
    const settings = new URLSearchParams(url.search).get('visible')
    const lower = (item) => {
        const lowerItem = item.toLowerCase()
        if (possibleFilters.includes(lowerItem)) {
            return lowerItem
        }
        return null
    }
    return settings === null ?
        possibleFilters :
        [...new Set(settings?.split(',').map(lower).filter((item) => item))]
}

const hideCategory = (categoryToHide) => {
    const url = new URL(window.location.href)
    const visibleParams = new URLSearchParams(url.search).get('visible')
    const currentVisible = visibleParams ? visibleParams.split(',') : [...possibleFilters]
    const settings = [...new Set(currentVisible)].filter((f) => f !== categoryToHide).join(',')

    url.searchParams.set('visible', settings)
    window.history.pushState({}, null, unescape(url.href))
}

const showCategory = (categoryToShow) => {
    if (typeof window === 'undefined') {
        return
    }
    const url = new URL(window.location.href)
    const currentVisible = new URLSearchParams(url.search).get('visible')?.split(',').filter(Boolean) ||
        [...possibleFilters]
    const settings = [...new Set([categoryToShow, ...currentVisible])]
    const noFilter = possibleFilters.length === settings.length || !settings.length

    noFilter ? url.searchParams.delete('visible') : url.searchParams.set('visible', settings.join(','))
    window.history.pushState({}, null, unescape(url.href))
}

const getSort = (initialSort) => {
    const url = new URL(window.location.href)
    let sort = new URLSearchParams(url.search).get('sort')
    if (!sort) {
        sort = initialSort || 'result'
    }
    return sort
}

const setSort = (type) => {
    const url = new URL(window.location.href)
    url.searchParams.set('sort', type)
    window.history.pushState({}, null, unescape(url.href))
}

const getCollapsedCategory = (renderCollapsed) => {
    let categories
    if (typeof window !== 'undefined') {
        const url = new URL(window.location.href)
        const collapsedItems = new URLSearchParams(url.search).get('collapsed')
        switch (true) {
        case !renderCollapsed && collapsedItems === null:
            categories = ['passed']
            break
        case collapsedItems?.length === 0 || /^["']{2}$/.test(collapsedItems):
            categories = []
            break
        case /^all$/.test(collapsedItems) || collapsedItems === null && /^all$/.test(renderCollapsed):
            categories = [...possibleFilters]
            break
        default:
            categories = collapsedItems?.split(',').map((item) => item.toLowerCase()) || renderCollapsed
            break
        }
    } else {
        categories = []
    }
    return categories
}

const getSortDirection = () => JSON.parse(sessionStorage.getItem('sortAsc')) || false
const setSortDirection = (ascending) => sessionStorage.setItem('sortAsc', ascending)

const getCollapsedIds = () => JSON.parse(sessionStorage.getItem('collapsedIds')) || []
const setCollapsedIds = (list) => sessionStorage.setItem('collapsedIds', JSON.stringify(list))

module.exports = {
    getVisible,
    hideCategory,
    showCategory,
    getCollapsedIds,
    setCollapsedIds,
    getSort,
    setSort,
    getSortDirection,
    setSortDirection,
    getCollapsedCategory,
    possibleFilters,
}

},{}]},{},[4]);
    </script>
  </footer>
</html>